store/mongo/migration_2_0_0.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2021 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package mongo
+
+import (
+	"context"
+
+	"github.com/mendersoftware/go-lib-micro/identity"
+	"github.com/mendersoftware/go-lib-micro/mongo/migrate"
+	mstorev1 "github.com/mendersoftware/go-lib-micro/store"
+	mstore "github.com/mendersoftware/go-lib-micro/store/v2"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	mopts "go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type migration_2_0_0 struct {
+	ms  *DataStoreMongo
+	ctx context.Context
+}
+
+func str2ptr(v string) *string {
+	return &v
+}
+
+func (m *migration_2_0_0) Up(from migrate.Version) error {
+	id := identity.FromContext(m.ctx)
+	if id != nil && id.Tenant != "" {
+		return nil
+	}
+
+	ctx := context.Background()
+	client := m.ms.client
+	dbs, err := migrate.GetTenantDbs(ctx, client, mstorev1.IsTenantDb(DbName))
+	if err != nil {
+		return err
+	}
+	dbs = append([]string{DbName}, dbs...)
+
+	collections := map[string]struct {
+		Indexes []mongo.IndexModel
+	}{
+		DbDevicesColl: {
+			Indexes: []mongo.IndexModel{
+				{
+					Keys: bson.D{
+						{Key: mstore.FieldTenantID, Value: 1},
+						{Key: DbDevId, Value: 1},
+					},
+					Options: &mopts.IndexOptions{
+						Name: str2ptr(mstore.FieldTenantID + "_" + DbDevId),
+					},
```

Comment:
Personally I prefer the functional declaration of Mongo options:
```suggestion
					Options: mopts.Index().
						SetName(mstore.FieldTenantID + "_" + DbDevId),
```